packages/docusaurus-theme-classic/src/theme/DocRoot/Layout/Sidebar/index.tsx

```diff
@@ -64,7 +68,10 @@ export default function DocRootLayoutSidebar({
           className={clsx(
             styles.sidebarViewport,
             hiddenSidebar && styles.sidebarViewportHidden,
-          )}>
+          )}
+          style={{
+            maxHeight: `calc(100vh - ${announcementBarHeightOnViewport}px)`,
```

Comment:
What if we had something more generic?

The theme could inline (like we do for dark mode) some JS logic that sets a `--scroll-Y` variable at the html root (could be useful in other cases 🤷‍♂️ 🤔 ) that we could use in a CSS calc/max/min/clamp expression here?

Somehow announcementBarHeightOnViewport = `clamp(0, var(--scroll-Y), var(--announcementBarHeight))`

Does it make sense?